### PR TITLE
Add a flag to enable grpc.

### PR DESCRIPTION
**Pull request by sbunce**
This is needed to get protoc-gen-go to output grpc code.

### CODE REVIEW

**Comment by dsymonds** [comment]
This will need a bit of thought. You're right that this is the required mechanism on the CodeGeneratorRequest edge, but I'm not sure I want to bake this in in this form. The specific `plugins=grpc` value only works with `protoc-gen-go`, and `gotoc` is trying to be agnostic to the generated language.

`protoc` handles this by overloading the `--go_out` flag. I'm not keen to follow that precedence. Maybe this should be a --params flag instead?

**Comment by sbunce** [comment]
I see your point. It does seem odd that protoc specifies plugins in the output flag value. My original solution of having a "-grpc" flag also doesn't seem quite right because we may want to specify additional plugins in the future.

I like your idea of "-params" because it allows us to specify additional plugins in the future without adding additional flags. I updated the pull request to use "-params".

No rush on this. This is an API change of sorts. Hard to change later.

File: main.go [hunk 1]
```diff
@@ -78,6 +79,9 @@ func main() {
 		FileToGenerate: flag.Args(),
 		ProtoFile:      fds.File,
 	}
+	if *params != "" {
+		cgRequest.Parameter = proto.String("plugins="+*params)
```
Comment (dsymonds) [inline main.go]: This should pass the flag value through unmodified.

```go
cgRequest.Parameter = params
```

File: main.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ var (
 	importPath     = flag.String("import_path", ".", "Comma-separated list of paths to search for imports.")
 	pluginBinary   = flag.String("plugin", "protoc-gen-go", "The code generator plugin to use.")
 	descriptorOnly = flag.Bool("descriptor_only", false, "Whether to print out only the FileDescriptorSet.")
+	params         = flag.String("params", "", "Parameters (comma separated values) to provide to the code generator.")
```
Comment (dsymonds) [inline main.go]: "Parameters to pass to the code generator plugin (plugin-specific format)."

**Review by dsymonds — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (sbunce) [inline main.go]: Done.

File: main.go (hunk 1 above)
Comment (sbunce) [inline main.go]: Done.

**Comment by sbunce** [comment]
I kept looking at things in an overly-specific way. "-grpc", then "plugins=". Passing the raw parameters does seem like the best way to go. Thanks!

Updated the pull request.

**Comment by dsymonds** [comment]
Merged in 0071f33. Thanks!

**Pull request closed without merging**